src/worker/jobs/downloads/process_log.rs

```diff
@@ -343,16 +343,16 @@ impl Debug for NameAndVersion {
 
 /// Checks if the given log file has already been processed.
 ///
-/// Calls [`spawn_blocking()`] and acquires a connection from the pool before
+/// Acquires a connection from the pool before
 /// passing it to the [`already_processed_inner()`] function.
-async fn already_processed(path: impl Into<String>, db_pool: DieselPool) -> anyhow::Result<bool> {
+async fn already_processed(path: impl Into<String>, db_pool: Pool) -> anyhow::Result<bool> {
     let path = path.into();
 
-    let already_processed = spawn_blocking(move || {
-        let mut conn = db_pool.get()?;
-        Ok::<_, anyhow::Error>(already_processed_inner(path, &mut conn)?)
-    })
-    .await?;
+    let conn = db_pool.get().await?;
+    let already_processed = conn
+        .interact(move |conn| already_processed_inner(path, conn))
+        .await
+        .map_err(|err| anyhow!(err.to_string()))??;
```

Comment:
Is it worth us providing a trait to wrap this pattern, since it's coming up in every job?

Reply:
you mean something like `.interact_anyhow(...)`?